packages/jest-runner/src/testWorker.ts

```diff
@@ -106,7 +107,28 @@ export async function worker({
       },
       sendMessageToJest,
     );
+    return makeSerializableTestResults(testResult);
   } catch (error: any) {
     throw formatError(error);
   }
 }
+
+function makeSerializableTestResults(result: TestResult): TestResult {
+  const {testResults} = result;
+  const serializableResults = testResults.map(resultItem => {
+    if (resultItem.failureDetails) {
```

Comment:
Do I get it right: the problem is that `failureDetails` includes data which cannot be serialised? If so, it be better to serialised it earlier in the test runner.

This was discussed already, I will try to dig out the thread. The idea is that we know in advance that all what the runner returns has to be serialisable, because that data will be passed through workers. If I remember it right, the test runner is already serialising most of the data. Seem like something was simply overlooked.

Reply:
> Do I get it right: the problem is that failureDetails includes data which cannot be serialised?

Yes, exactly, the problem with `function` type which throw an error during clone. Also DOM nodes might cause similar behaviour, but I'm not sure if it's a valid case for workers. 

> If so, it be better to serialised it earlier in the test runner.

I was also thinking about it, my only concern was that it might be exposed to other tools (outside of `jest` repo). But if the approach is ok I can fix this and add some tests to cover the serialisation

> This was discussed already, I will try to dig out the thread

Sounds good, found [this related PR](https://github.com/jestjs/jest/pull/10989), would be nice to look into other discussions

Reply:
@mrazauskas Thanks a lot, now I have full context